src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -452,4 +459,35 @@ public void setVelocity(Vector3d velocity) {
     public void setVelocityOverride(@Nullable Vector3d velocity) {
         this.velocityOverride = velocity;
     }
+
+    /**
+     * @author kashike - 30/12/2015
+     */
+    @Overwrite
+    public void setGameType(WorldSettings.GameType gameType) {
+        // Sponge start
+        ChangeGameModeEvent.TargetPlayer event = SpongeEventFactory.createChangeGameModeEventTargetPlayer(Cause.of(NamedCause.source(this)),
```

Comment:
I'm fairly sure this can be done with either a `ModifyArgs` or `Inject` with capture locals.